app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -2,7 +2,23 @@
 
 module Users
   class OmniauthCallbacksController < Devise::OmniauthCallbacksController
-    layout 'authentication', only: %i[new create]
+    skip_before_action :verify_authenticity_token, only: :github
+
+    def github
+      @user = User.from_omniauth(request.env['omniauth.auth'])
+
+      if @user.persisted?
+        sign_in_and_redirect @user, event: :authentication
+        set_flash_message(:notice, :success, kind: 'Github') if is_navigational_format?
+      else
+        session['devise.github_data'] = request.env['omniauth.auth'].except!(:extra)
+        redirect_to new_user_registration_url
+      end
+
+      # # puts('GITHUB ACTION')
```

Comment:
Don't forget to remove old comments and debugging code 😄

Reply:
Will do, thanks for catching that!